rocketpy/sensors/gyroscope.py

```diff
@@ -0,0 +1,230 @@
+            The random walk of the sensor in rad/s/√Hz. Sometimes called "bias
+            (in)stability" or "bias drift"". Default is 0, meaning no random
+            walk is applied. If a float or int is given, the same random walk is
+            applied to all axes. The values of each axis can be set individually
+            by passing a list of length 3.
+        constant_bias : float, list, optional
+            The constant bias of the sensor in rad/s. Default is 0, meaning no
+            constant bias is applied. If a float or int is given, the same bias
+            is applied to all axes. The values of each axis can be set
+            individually by passing a list of length 3.
+        operating_temperature : float, optional
+            The operating temperature of the sensor in degrees Celsius. At 25°C,
+            the temperature bias and scale factor are 0. Default is 25.
+        temperature_sensitivity : float, list, optional
+            The temperature bias of the sensor in rad/s/°C. Default is 0,
+            meaning no temperature bias is applied. If a float or int is given,
+            the same temperature bias is applied to all axes. The values of each
+            axis can be set individually by passing a list of length 3.
+        temperature_scale_factor : float, list, optional
+            The temperature scale factor of the sensor in %/°C. Default is 0,
+            meaning no temperature scale factor is applied. If a float or int is
+            given, the same temperature scale factor is applied to all axes. The
+            values of each axis can be set individually by passing a list of
+            length 3.
+        cross_axis_sensitivity : float, optional
+            Skewness of the sensor's axes in percentage. Default is 0, meaning
+            no cross-axis sensitivity is applied.
+            of each axis can be set individually by passing a list of length 3.
+        acceleration_sensitivity : float, list, optional
+            Sensitivity of the sensor to linear acceleration in rad/s/g. Default
+            is 0, meaning no sensitivity to linear acceleration is applied. If a
+            float or int is given, the same sensitivity is applied to all axes.
+            The values of each axis can be set individually by passing a list of
+            length 3.
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(
+            sampling_rate,
+            orientation,
+            measurement_range=measurement_range,
+            resolution=resolution,
+            noise_density=noise_density,
+            random_walk=random_walk,
+            constant_bias=constant_bias,
+            operating_temperature=operating_temperature,
+            temperature_bias=temperature_bias,
+            temperature_scale_factor=temperature_scale_factor,
+            cross_axis_sensitivity=cross_axis_sensitivity,
+            name=name,
+        )
+        self.type = "Gyroscope"
```

Comment:
I believe this is the only code issue I have with this PR.

It seems an antipattern to me having in each concrete object a common variable that is not in the abstract one to refer to the sensor type. I understand this might speed up some is instance checks, but it is odd for a instance to store its own type.

What is the reasoning behind this argument and where is it used?

Reply:
fixed in https://github.com/RocketPy-Team/RocketPy/pull/583/commits/de2d8bd46332c1ffaff890e8b07bcdad44b0729d